src/hotspot/share/runtime/perfData.hpp

```diff
@@ -830,12 +831,18 @@ class PerfTraceTime : public StackObj {
     PerfLongCounter* _timerp;
 
   public:
-    inline PerfTraceTime(PerfLongCounter* timerp) : _timerp(timerp) {
-      if (!UsePerfData) return;
+    inline PerfTraceTime(PerfLongCounter* timerp, bool is_on = true) : _timerp(timerp) {
+      if (!is_on || !UsePerfData) return;
```

Comment:
Instead of having a separate `is_on` parameter, can we check for `timerp == nullptr1` instead?

Reply:
It works. I've pushed another commit.